src/crypto/CHIPCryptoPAL.h

```diff
@@ -1492,6 +1492,27 @@ CHIP_ERROR ExtractSKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan
  **/
 CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan & akid);
 
+/**
+ * @brief Checks for resigned version of the certificate in the list and returns it.
+ *
+ * The following conditions SHOULD be satisfied for the certificate to qualify as
+ * a resigned version of a reference certificate:
+ *   - SKID of the candidate and the reference certificate should match.
+ *   - SubjectDN of the candidate and the reference certificate should match.
+ *
+ * If no resigned version is found then reference certificate itself is returned.
+ *
+ *  @param referenceCertificate       A DER certificate.
+ *  @param candidateCertificates      A pointer to the list of DER Certificates, which should be searched
+ *                                    for the resigned version of `referenceCertificate`.
+ *  @param candidateCertificatesCount Number of certificates in the `candidateCertificates` list.
+ *  @param outCertificate             A reference to the certificate or it's resigned version if found.
```

Comment:
Please note that it points to either referenceCertificate or one of candidateCertificates, and it's not a copy.

Reply:
added clarification